### PR TITLE
[PHEE-697] Add Payee dfsp id header in batch transaction API

### PR DESCRIPTION
**Pull request by apurbraj**
Sending payee dfspid variable to closedloop workflow in iniSubBatch
Pending Task - Validating payee dfspId [PHEE-698](https://fynarfin.atlassian.net/browse/PHEE-698)

## Description

* PR title should have jira ticket enclosed in `[]`.<br>
Format: ``` [jira_ticket] description```<br>
ex: [phee-123] PR title.
* Add a link to the Jira ticket.
* Describe the changes made and why they were made.

## Checklist

Please make sure these boxes are checked before submitting your pull request - thanks!
- [ ] Followed the PR title naming convention mentioned above.

- [ ] Design related bullet points or design document link related to this PR added in the description above.

- [ ] Updated corresponding Postman Collection or Api documentation for the changes in this PR.

- [ ] Created/updated unit or integration tests for verifying the changes made.

- [ ] Added required Swagger annotation and update API documentation with details of any API changes if applicable

- [ ] Followed the naming conventions as given in https://docs.google.com/document/d/1Q4vaMSzrTxxh9TS0RILuNkSkYCxotuYk1Xe0CMIkkCU/edit?usp=sharing

[PHEE-698]: https://fynarfin.atlassian.net/browse/PHEE-698?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/main/java/org/mifos/processor/bulk/api/definition/BatchTransactions.java [hunk 1]
```diff
@@ -24,6 +25,7 @@ String batchTransactions(HttpServletRequest httpServletRequest, HttpServletRespo
             @RequestHeader(value = HEADER_TYPE) String type, @RequestHeader(value = HEADER_PLATFORM_TENANT_ID) String tenant,
             @RequestHeader(value = HEADER_REGISTERING_INSTITUTE_ID, required = false) String registeringInstitutionId,
             @RequestHeader(value = HEADER_PROGRAM_ID, required = false) String programId,
-            @RequestHeader(value = CALLBACK, required = false) String callbackUrl) throws IOException;
+            @RequestHeader(value = CALLBACK, required = false) String callbackUrl,
+            @RequestHeader(value = PAYEE_DFSP_ID, required = false) String payeeDFSPId) throws IOException;
```
Comment (fynmanoj) [inline src/main/java/org/mifos/processor/bulk/api/definition/BatchTransactions.java:29]: update postman

File: src/main/java/org/mifos/processor/bulk/camel/config/CamelProperties.java [hunk 2]
```diff
@@ -85,4 +85,5 @@ private CamelProperties() {}
     public static final String ORIGINAL_TRANSACTION_LIST = "originalTransactionList";
     public static final String CALLBACK = "X-CallbackURL";
     public static final String CONTENT_TYPE = "Content-Type";
+    public static final String PAYEE_DFSP_ID = "X-PayeeDFSP-ID";
```
Comment (fynmanoj) [inline src/main/java/org/mifos/processor/bulk/camel/config/CamelProperties.java:88]: use this naming convention only for API header. don't use it for Zeebe variable

File: src/main/java/org/mifos/processor/bulk/camel/routes/InitSubBatchRoute.java [hunk 3]
```diff
@@ -118,6 +125,18 @@ public void configure() throws Exception {
...
                     variables.put(DEBULKINGDFSPID, mapping.getDebulkingDfspid() == null ? tenantName : mapping.getDebulkingDfspid());
+                    if (!(Boolean) variables.get(PARTY_LOOKUP_FAILED)) {
+                        String filename = exchange.getProperty(SERVER_FILE_NAME).toString();
+                        String regex = ".*_sub-batch-([\\w-]+)\\.csv";
+                        Pattern pattern = Pattern.compile(regex);
+                        Matcher matcher = pattern.matcher(filename);
+
+                        if (matcher.matches()) {
+                            String payeeDFSPId = matcher.group(1);
+                            logger.info("Payee DFSP Id {}", payeeDFSPId);
```
Comment (fynmanoj) [inline src/main/java/org/mifos/processor/bulk/camel/routes/InitSubBatchRoute.java:136]: set log level as debug

File: src/main/java/org/mifos/processor/bulk/camel/routes/ProcessorStartRoute.java [hunk 4]
```diff
@@ -131,12 +132,14 @@ private void setup() {
             String requestId = exchange.getIn().getHeader("requestId", String.class);
             String purpose = exchange.getIn().getHeader("purpose", String.class);
             String batchId = UUID.randomUUID().toString();
+            String payeeDFSPId = exchange.getIn().getHeader("X-PayeeDFSP-ID", String.class);
```
Comment (fynmanoj) [inline src/main/java/org/mifos/processor/bulk/camel/routes/ProcessorStartRoute.java:135]: use constant here

File: src/main/java/org/mifos/processor/bulk/connectors/service/ProcessorStartRouteService.java [hunk 5]
```diff
@@ -248,6 +250,7 @@ public void startBatchProcessCsv(Exchange exchange) throws IOException {
         variables.put(PAYER_IDENTIFIER_VALUE, exchange.getProperty(PAYER_IDENTIFIER_VALUE));
         variables.put(REGISTERING_INSTITUTE_ID, exchange.getProperty(REGISTERING_INSTITUTE_ID));
         variables.put(IS_FILE_VALID, true);
+        variables.put(PAYEE_DFSP_ID, payeeDFSPId);
```
Comment (fynmanoj) [inline src/main/java/org/mifos/processor/bulk/connectors/service/ProcessorStartRouteService.java:253]: use this naming convention only for API header. don't use it for Zeebe variable

File: src/main/java/org/mifos/processor/bulk/zeebe/ZeebeVariables.java [hunk 6]
```diff
@@ -165,5 +165,6 @@ private ZeebeVariables() {}
     public static final String PARTY_LOOKUP_SUCCESSFUL_TRANSACTION_AMOUNT = "totalApprovedAmount";
     public static final String PARTY_LOOKUP_SUCCESSFUL_TRANSACTION_COUNT = "totalApprovedCount";
     public static final String REGISTERING_INSTITUTION_ID = "X-Registering-Institution-ID";
+    public static final String PAYEE_DFSP_ID = "X-PayeeDFSP-ID";
```
Comment (fynmanoj) [inline src/main/java/org/mifos/processor/bulk/zeebe/ZeebeVariables.java:168]: use this naming convention only for API header. don't use it for Zeebe variable

File: src/main/java/org/mifos/processor/bulk/camel/routes/InitSubBatchRoute.java [hunk 7]
```diff
@@ -118,6 +125,18 @@ public void configure() throws Exception {
                     Map<String, Object> variables = exchange.getProperty(ZEEBE_VARIABLE, Map.class);
                     variables.put(PAYMENT_MODE, paymentMode);
                     variables.put(DEBULKINGDFSPID, mapping.getDebulkingDfspid() == null ? tenantName : mapping.getDebulkingDfspid());
+                    if (!(Boolean) variables.get(PARTY_LOOKUP_FAILED)) {
+                        String filename = exchange.getProperty(SERVER_FILE_NAME).toString();
+                        String regex = ".*_sub-batch-([\\w-]+)\\.csv";
```
Comment (fynmanoj) [inline src/main/java/org/mifos/processor/bulk/camel/routes/InitSubBatchRoute.java:130]: this needs to be added to the documentation. We might miss how payee fsp is transferred

**Review by fynmanoj — CHANGES_REQUESTED** [review]

File: src/main/java/org/mifos/processor/bulk/connectors/service/ProcessorStartRouteService.java (hunk 5 above)
Comment (apurbraj) [reply src/main/java/org/mifos/processor/bulk/connectors/service/ProcessorStartRouteService.java:253]: Done

**Review by fynmanoj — APPROVED** [review]

**Pull request merged**